Error: Cannot find module '/xxxxxxx/node_modules/chai/register-should'

Hi :)

I'm facing as is the title issue.

## Env & Phenomenon

I confirmed this problem happened in chai `4.2.0`, `4.6.3` and Node.js `12.x`, `14.x`, `16.x`. I do not check other chai and node.js versions.

I'm using chai with mocha and I specified `chai/register-should` in `mocha.rc.yml`.

```yml
// mocha.rc.yml
require: 
  - "chai/register-should"
```

Also this problem occur with `mocha spec.js -r chai/register-should ` command.

## Error log

```js
✖ ERROR: Error: Cannot find module '/home/runner/work/hexo/hexo/node_modules/chai/register-should'
    at createEsmNotFoundErr (internal/modules/cjs/loader.js:929:15)
    at finalizeEsmResolution (internal/modules/cjs/loader.js:922:15)
    at resolveExports (internal/modules/cjs/loader.js:450:[14](https://github.com/hexojs/hexo/runs/5167084831?check_suite_focus=true#step:5:14))
    at Function.Module._findPath (internal/modules/cjs/loader.js:490:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:[18](https://github.com/hexojs/hexo/runs/5167084831?check_suite_focus=true#step:5:18))
    at exports.requireOrImport (/home/runner/work/hexo/hexo/node_modules/mocha/lib/nodejs/esm-utils.js:60:[20](https://github.com/hexojs/hexo/runs/5167084831?check_suite_focus=true#step:5:20))
    at async exports.handleRequires (/home/runner/work/hexo/hexo/node_modules/mocha/lib/cli/run-helpers.js:94:28)
    at async /home/runner/work/hexo/hexo/node_modules/mocha/lib/cli/run.js:353:[25](https://github.com/hexojs/hexo/runs/5167084831?check_suite_focus=true#step:5:25) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/runner/work/hexo/hexo/node_modules/chai/package.json'
}
npm ERR! Test failed.  See above for more details.
```

## Others

This issue seems fixed in https://github.com/chaijs/chai/commit/5276af683ca4f6699768b4401dcf4c7c331e83f7. Because, I specify the commit in `package.json` like below this problem is solved.

```json
"devDependencies": {
  "chai": "git+https://github.com/chaijs/chai.git#5276af683ca4f6699768b4401dcf4c7c331e83f7",
```

But, it commit reverted in https://github.com/chaijs/chai/commit/022c2faefc9a86b478392594c9015c3b942892e2. So, how to solve it?

Thank you :)

--- a/README.md
+++ b/README.md
@@ -98,9 +98,9 @@ import { expect, use } from 'chai';  // Creates local variables `expect` and `us
 ### Usage with Mocha
 
 ```bash
-mocha spec.js -r chai/register-assert.js  # Using Assert style
-mocha spec.js -r chai/register-expect.js  # Using Expect style
-mocha spec.js -r chai/register-should.js  # Using Should style
+mocha spec.js --require chai/register-assert.js  # Using Assert style
+mocha spec.js --require chai/register-expect.js  # Using Expect style
+mocha spec.js --require chai/register-should.js  # Using Should style
 ```
 
 [Read more about these styles in our docs](http://chaijs.com/guide/styles/).

--- a/README.md
+++ b/README.md
@@ -98,9 +98,9 @@ import { expect, use } from 'chai';  // Creates local variables `expect` and `us
 ### Usage with Mocha
 
 ```bash
-mocha spec.js -r chai/register-assert  # Using Assert style
-mocha spec.js -r chai/register-expect  # Using Expect style
-mocha spec.js -r chai/register-should  # Using Should style
+mocha spec.js -r chai/register-assert.js  # Using Assert style
+mocha spec.js -r chai/register-expect.js  # Using Expect style
+mocha spec.js -r chai/register-should.js  # Using Should style
 ```
 
 [Read more about these styles in our docs](http://chaijs.com/guide/styles/).

